File: backend/src/ProductManager.Domain/Entities/Product.cs
namespace ProductManager.Domain.Entities;

public class Product
{
 public Guid Id { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string? Descripcion { get; set; }
    public decimal Precio { get; set; }
    public bool Estado { get; set; }
    public string UsuarioCreacion { get; set; } = string.Empty;
    public DateTime FechaCreacion { get; set; }
    public string UsuarioModificacion { get; set; } = string.Empty;
    public DateTime FechaModificacion { get; set; }
}

File: frontend/src/types/product.ts
export interface Product {
  id: string;
  nombre: string;
  descripcion?: string;
  precio: number;
  estado: boolean;
  usuarioCreacion: string;
  fechaCreacion: string;
  usuarioModificacion: string;
  fechaModificacion: string;
}

export interface ProductListResponse {
  data: Product[];
  total: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
}

export interface CreateProductRequest {
  nombre: string;
  descripcion?: string;
  precio: number;
  estado: boolean;
}

export type UpdateProductRequest = CreateProductRequest;

export interface ProductFilters {
  nombre?: string;
  estado?: string;
  minPrecio?: string;
  maxPrecio?: string;
  pageNumber?: number;
  pageSize?: number;
}
